File: packages/ui/android/src/main/java/expo/modules/youversion/scriptureparagraph/YouVersionScriptureParagraphModule.kt
package expo.modules.youversion.scriptureparagraph

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class YouVersionScriptureParagraphModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("YouVersionScriptureParagraph")

    View(YouVersionScriptureParagraphView::class) {
      Events("onSizeChange", "onVersePress", "onFootnotePress")

      Prop("runs") { view: YouVersionScriptureParagraphView, runs: List<ScriptureRun> ->
        view.setRuns(runs)
      }
      Prop("firstIndent") { view: YouVersionScriptureParagraphView, value: Double ->
        view.setFirstIndent(value)
      }
      Prop("restIndent") { view: YouVersionScriptureParagraphView, value: Double ->
        view.setRestIndent(value)
      }
      Prop("lineHeightMultiple") { view: YouVersionScriptureParagraphView, value: Double ->
        view.setLineHeightMultiple(value)
      }
    }
  }
}

File: packages/ui/ios/YouVersionScriptureParagraphModule.swift
import ExpoModulesCore

public class YouVersionScriptureParagraphModule: Module {
  public func definition() -> ModuleDefinition {
    Name("YouVersionScriptureParagraph")

    View(YouVersionScriptureParagraphView.self) {
      Events("onSizeChange", "onVersePress", "onFootnotePress")

      Prop("runs") { (view: YouVersionScriptureParagraphView, runs: [ScriptureRun]) in
        view.update(runs: runs)
      }
      Prop("firstIndent") { (view: YouVersionScriptureParagraphView, value: Double) in
        view.update(firstIndent: value)
      }
      Prop("restIndent") { (view: YouVersionScriptureParagraphView, value: Double) in
        view.update(restIndent: value)
      }
      Prop("lineHeightMultiple") { (view: YouVersionScriptureParagraphView, value: Double) in
        view.update(lineHeightMultiple: value)
      }
    }
  }
}
